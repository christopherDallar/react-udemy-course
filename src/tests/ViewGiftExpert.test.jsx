const { render, screen } = require('@testing-library/react')
const { ViewGiftExpert } = require('../views/GiftExpert.view')

describe('Testing <ViewGiftExpert />', () => {
  test('should math with snapshot', () => {
    const { container } = render(<ViewGiftExpert />)
    expect(container).toMatchSnapshot()
  })

  test('should show title h1 tag GiftExpertApp', () => {
    const { container } = render(<ViewGiftExpert />)

    const h1 = container.querySelector('h1')

    expect(h1.innerHTML).toBe('GiftExpertApp')
  })
})
