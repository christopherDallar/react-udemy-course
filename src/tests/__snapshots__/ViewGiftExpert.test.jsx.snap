// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ViewGiftExpert /> should math with snapshot 1`] = `
<div>
  <h1>
    GiftExpertApp
  </h1>
  <form
    role="myForm"
  >
    <h1 />
    <input
      type="text"
      value=""
    />
  </form>
  <hr />
  <ol>
    <h3
      class="animate__animated animate__fadeIn"
    >
      One punch
    </h3>
    <p
      class="animate__animated animate__flash"
    >
      Loading
    </p>
    <div
      class="card-grid"
    />
  </ol>
</div>
`;
